cdm/src/main/java/ucar/nc2/ft2/coverage/CoverageDatasetFactory.java

```diff
@@ -126,17 +95,36 @@ static public FeatureDatasetCoverage open(String endpoint) throws IOException {
    * @throws IOException
    */
   static public Optional<FeatureDatasetCoverage> openGrib(String endpoint) throws IOException {
+
+    List<Object> notGribThrowables = Arrays.asList(
+            IllegalAccessException.class,
+            IllegalArgumentException.class,
+            ClassNotFoundException.class,
+            NoSuchMethodException.class,
+            NoSuchMethodError.class);
+
     try {
       Class<?> c = CoverageDatasetFactory.class.getClassLoader().loadClass("ucar.nc2.grib.coverage.GribCoverageDataset");
       Method method = c.getMethod("open", String.class);
       return (Optional<FeatureDatasetCoverage>) method.invoke(null, endpoint);
+    } catch (Exception e) {
+      for (Object noGrib : notGribThrowables) {
+        // check for possible errors that are due to the file not being grib. Need to look
+        // at the error causes too, as reflection error can be buried under a InvorcationTargetException
+        boolean notGribTopLevel = e.getClass().equals(noGrib);
+        boolean notGribBuried = e.getClass().equals(InvocationTargetException.class) &&
+                                e.getCause() != null &&
+                                e.getCause().getClass().equals(noGrib);
 
-    } catch (IllegalAccessException | IllegalArgumentException | ClassNotFoundException | NoSuchMethodException e) {
-      //e.printStackTrace();
-      return Optional.empty(NO_GRIB_CLASS);
+        if (notGribTopLevel || notGribBuried) {
+          return Optional.empty(NO_GRIB_CLASS);
+        } else if (e.getClass().equals(InvocationTargetException.class) && e.getCause() != null && e.getCause().getClass().equals(noGrib)) {
```

Comment:
I'm referring to the `else if` statement. That wasn't as clear as I thought it would be in the review that GitHub created.